tools/reviewer/service/tests/CodeReviewServiceGetDiffFilesTest.java

```diff
@@ -0,0 +1,450 @@
+
+  @Singleton
+  @Component(modules = {CommonModule.class, AaModule.class})
+  interface TestComponent {
+    AuthService getAuthService();
+
+    GitRepoFactory getGitRepoFactory();
+
+    InitCommand getInitCommand();
+
+    WorkspaceCommand getWorkspaceCommand();
+
+    FileUtils getFileUtils();
+
+    TextDifferencer getTextDifferencer();
+  }
+
+  private void createInitialRepo(String initialRepoFolder) {
+    fileUtils.mkdirs(initialRepoFolder);
+    GitRepo repo = gitRepoFactory.create(initialRepoFolder);
+    repo.init();
+    repo.setFakeUsersData();
+    fileUtils.writeStringUnchecked(
+        TEST_FILE_CONTENTS, fileUtils.joinPaths(initialRepoFolder, FILE_IN_HEAD));
+    fileInHeadCommitId = repo.commit(repo.getUncommittedFiles(), "Initial commit").getId();
+  }
+
+  private void initAaBase(String initialRepoFolder, String aaBaseFolder) {
+    InitCommand initCommand = component.getInitCommand();
+    InitCommand.startuposRepo.resetValueForTesting();
+    String[] args = {
+      "init", aaBaseFolder,
+      "--startupos_repo", initialRepoFolder,
+    };
+    initCommand.run(args);
+  }
+
+  private void createAaWorkspace(String name) {
+    WorkspaceCommand workspaceCommand = component.getWorkspaceCommand();
+    String[] args = {"workspace", "-f", name};
+    workspaceCommand.run(args);
+    repoPath = fileUtils.joinPaths(getWorkspaceFolder(TEST_WORKSPACE), "startup-os");
+    repo = gitRepoFactory.create(repoPath);
+    repo.setFakeUsersData();
+    repo.switchBranch("D" + DIFF_ID);
+  }
+
+  private void createBlockingStub() throws IOException {
+    // Generate a unique in-process server name.
+    String serverName = InProcessServerBuilder.generateName();
+    server =
+        InProcessServerBuilder.forName(serverName)
+            .directExecutor()
+            .addService(codeReviewService)
+            .build()
+            .start();
+    channel = InProcessChannelBuilder.forName(serverName).directExecutor().build();
+    blockingStub = CodeReviewServiceGrpc.newBlockingStub(channel);
+  }
+
+  private String joinPaths(String first, String... more) {
+    return FileSystems.getDefault().getPath(first, more).toAbsolutePath().toString();
+  }
+
+  private String getWorkspaceFolder(String workspace) {
+    return joinPaths(aaBaseFolder, "ws", workspace);
+  }
+
+  private void writeFile(String contents) {
+    writeFile(TEST_FILE, contents);
+  }
+
+  private void writeFile(String filename, String contents) {
+    fileUtils.writeStringUnchecked(
+        contents, fileUtils.joinPaths(getWorkspaceFolder(TEST_WORKSPACE), "startup-os", filename));
+  }
+
+  private Protos.DiffFilesResponse getResponse() {
+    Protos.DiffFilesRequest request =
+        Protos.DiffFilesRequest.newBuilder()
+            .setWorkspace(TEST_WORKSPACE)
+            .setDiffId(DIFF_ID)
+            .build();
+
+    return blockingStub.getDiffFiles(request);
+  }
+
+  private com.google.startupos.common.repo.Protos.BranchInfo getExpectedBranchInfo() {
```

Comment:
It's better to import `BranchInfo` instead. That way all imports are at the top and someone looking there can get an idea of what this file is using.
Similarly with other protos.

Reply:
Done.